# ancient_MAG_pipeline

A shell script pipeline intended to aid in the creation of **M**etagenomic **A**ssembled **G**enomes (**MAGs**) from ancient microbiome samples. The paper associated with this data is available here: [Standeven et al.]().

## Pipeline software pre-requisites

- **Megahit v1.2.9**: A tool used to assemble large and complex *de novo* datasets [(Li et al. 2015)](https://pubmed.ncbi.nlm.nih.gov/25609793/). Can be accessed/downloaded [here](https://github.com/voutcn/megahit).

- **MetaBAT v2.12.1**: A tool used for contig binning ([Kang et al. 2015])(https://peerj.com/articles/1165/). Can be accessed/downloaded [here](https://bitbucket.org/berkeleylab/metabat/src/master/).

- **Maxbin2 v2.2.7**: A tool used for contig binning [(Wu et al. 2016)](https://academic.oup.com/bioinformatics/article/32/4/605/1744462). Can be accessed/downloaded [here](https://sourceforge.net/projects/maxbin2/).

- **CONCOCT v1.1.0**: A contig binning tool [(Alneberg et al. 2014)](https://www.nature.com/articles/nmeth.3103). Can be accessed/downloaded [here](https://github.com/BinPro/CONCOCT).

- **DAStool v1.1.6**: A tool to select high qualtiy bins ([Sieber et al. 2018])(https://www.nature.com/articles/s41564-018-0171-1). Can be accessed/downloaded [here](https://github.com/cmks/DAS_Tool).

- **CheckM v1.2.2**: A tool to assess completeness and contamination of MAGs ([Parks et al. 2015])(http://genome.cshlp.org/content/25/7/1043.short). Can be accessed/downloaded [here](https://github.com/Ecogenomics/CheckM).

- **GTDBTK**: text

- **Bowtie2 v2.5.0**: A tool for read mapping sequences to a reference genome [(Langmead and Salzberg, 2012)](https://academic.oup.com/bioinformatics/article/35/3/421/5055585?login=false). Can be accessed/downloaded [here](https://github.com/BenLangmead/bowtie2).

- **SAMtools**: text

## Considerations before beginning

Text...
